src/router/index.js

```diff
@@ -9,6 +12,41 @@ const routes = [
     path: "/",
     name: "Home",
     component: Home,
+    children: [
+      {
+        path: "list/:listId",
+        name: "ListDetails",
+        component: ListDetails,
+        props: (route) => {
+          const allTasksByListId = store.tasksData.filter(
+            (task) => task.id_list === route.params.listId
+          );
+          return {
+            listName: store.listsData.find(
+              (list) => list.id === route.params.listId
+            ).name,
+            allTasks: [...allTasksByListId],
+          };
+        },
+        beforeEnter: (to, from, next) => {
+          // fetching list data goes here
+          const listsData = store.getAllLists();
+          // check if list id exist in lists data
+          const exists = listsData.find((list) => list.id === to.params.listId);
+          if (exists) {
+            return next();
+          } else {
+            return next({ name: "PageNotFound" });
+          }
+        },
+      },
+    ],
+  },
+  {
+    path: "/404",
```

Comment:
The `PageNotFound` should appear whenever the URL doesn't match any of the exact URLs in our routes.
```suggestion
    path: "*",
```

Reply:
Updated!

Reply:
Sorry for the incorrect suggestion above as it may lead to an unexpected error when clicking on a newly created list. Please revert the change and refer to [this guide](https://github.com/vuejs/vue-router/issues/724#issuecomment-285029091) if needed.

Reply:
Ok, Updated!